File: python/src/workflows.py
from datetime import timedelta
from temporalio import workflow
from activities import sayName


@workflow.defn
class HelloWorldWorkflow:
    @workflow.run
    async def run(self, name: str) -> str:
        return await workflow.execute_activity(
            sayName,
            name,
            schedule_to_close_timeout=timedelta(seconds=10),
        )

File: typescript/src/workflows.ts
import { proxyActivities } from '@temporalio/workflow';

import type { createActivities } from './activities';

const { sayName } = proxyActivities<ReturnType<typeof createActivities>>({
  startToCloseTimeout: '1 minute',
});

// Add Workflow Definitions here.
export async function helloWorldWorkflow(name: string): Promise<string> {
  // Your workflow code here
  return sayName(name);
}
